packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -407,62 +406,57 @@ export const Audio = () => {
       ? queueTracks.slice(refUids.length)
       : queueTracks
 
-    const newTrackData = await Promise.all(
-      newQueueTracks.map(async (track) => {
-        const trackOwner = queueTrackOwnersMap[track.owner_id]
-        const trackId = track.track_id
-        const offlineTrackAvailable =
-          trackId &&
-          isOfflineModeEnabled &&
-          offlineAvailabilityByTrackId[trackId]
-
-        // Get Track url
-        let url: string
-        let isM3u8 = false
-        if (offlineTrackAvailable && isCollectionMarkedForDownload) {
-          const audioFilePath = getLocalAudioPath(trackId)
-          url = `file://${audioFilePath}`
-        } else if (isStreamMp3Enabled && isReachable) {
-          url = apiClient.makeUrl(
-            `/tracks/${encodeHashId(track.track_id)}/stream`
-          )
-        } else {
-          isM3u8 = true
-          const ownerGateways =
-            audiusBackendInstance.getCreatorNodeIPFSGateways(
-              trackOwner.creator_node_endpoint
-            )
-          url = hlsUtils.generateM3U8Variants({
-            segments: track?.track_segments ?? [],
-            gateways: ownerGateways
-          })
-        }
-
-        const localTrackImageSource =
-          isNotReachable && track
-            ? { uri: `file://${getLocalTrackCoverArtPath(trackId.toString())}` }
-            : undefined
-
-        const imageUrl =
-          getImageSourceOptimistic({
-            cid: track ? track.cover_art_sizes || track.cover_art : null,
-            user: trackOwner,
-            size: SquareSizes.SIZE_1000_BY_1000,
-            localSource: localTrackImageSource
-          })?.uri ?? DEFAULT_IMAGE_URL
+    const newTrackData = newQueueTracks.map((track) => {
+      const trackOwner = queueTrackOwnersMap[track.owner_id]
+      const trackId = track.track_id
+      const offlineTrackAvailable =
+        trackId && isOfflineModeEnabled && offlineAvailabilityByTrackId[trackId]
+
+      // Get Track url
+      let url: string
+      let isM3u8 = false
+      if (offlineTrackAvailable && isCollectionMarkedForDownload) {
+        const audioFilePath = getLocalAudioPath(trackId)
+        url = `file://${audioFilePath}`
+      } else if (isStreamMp3Enabled && isReachable) {
+        url = apiClient.makeUrl(
+          `/tracks/${encodeHashId(track.track_id)}/stream`
+        )
+      } else {
+        isM3u8 = true
+        const ownerGateways = audiusBackendInstance.getCreatorNodeIPFSGateways(
+          trackOwner.creator_node_endpoint
+        )
+        url = hlsUtils.generateM3U8Variants({
+          segments: track?.track_segments ?? [],
```

Comment:
nit: seems like you don't need the `?` as you haven't used it in the lines of code above that use `track`

Reply:
unless the lines above are the ones that need the `?` to check for null/undefined?